src/App.tsx

```diff
@@ -1,34 +1,97 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+export type FilterByCompletion = 'all' | 'active' | 'completed';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | null | undefined>(null);
+  const [selectedTask, setSelectedTask] = useState<number | null>(null);
+  const [isLoaded, setIsLoaded] = useState<boolean>(false);
+  const [completeStatus, setCompleteStatus] = useState<string>('all');
+  const [searchInput, setSearchInput] = useState<string>('');
+  const [error, setError] = useState('');
+
+  const todo = todos?.find(t => t.id === selectedTask);
+  let filteredTodos = todos;
+
+  switch (completeStatus) {
+    case 'active':
+      filteredTodos = todos?.filter(task => task.completed === false);
+      break;
+    case 'completed':
+      filteredTodos = todos?.filter(task => task.completed === true);
+      break;
+    case 'all':
+      filteredTodos = todos;
+      break;
+    default:
+      break;
+  }
```

Comment:
You can simply use `useMemo` here
```suggestion
const filteredTodos = useMemo(() => {
  const normalizedSearch = searchInput.toLowerCase().trim();
  
  let filtered = todos;
  
  switch (completeStatus) {
    case 'active':
      filtered = filtered?.filter(task => !task.completed);
      break;
    case 'completed':
      filtered = filtered?.filter(task => task.completed);
      break;
    case 'all':
      break;
    default:
      throw new Error("Invalid completeStatus");
  }
  
  if (!normalizedSearch) {
    return filtered;
  }
    
  return filtered?.filter(todo => 
    todo.title?.toLowerCase().trim().includes(normalizedSearch)
  );
}, [todos, completeStatus, searchInput]);
```